cmd_clean/main.go

```diff
@@ -53,10 +55,57 @@ func (t *Transport) base() http.RoundTripper {
 	return http.DefaultTransport
 }
 
+// Convert certificate in bytes array into stripped header for envoy sidecar
+func certPemBytesToHeader(caFile string) string {
+	certPem, err := os.ReadFile(caFile)
+	if err != nil {
+		log.Error(err, "failed to read ca file")
+		return ""
+	}
+	cert := string(certPem)
+	cert = strings.ReplaceAll(cert, "-----BEGIN CERTIFICATE-----", "")
+	cert = strings.ReplaceAll(cert, "-----END CERTIFICATE-----", "")
+	cert = strings.ReplaceAll(cert, "\n", "")
+	return strings.TrimSpace(cert)
+}
+
+// just a demo to show how to use customer http client
+// customer http client should handle verify and authentication
+// here using the basic user/password mode for authentication
+// not handling verify for none envoy client
+func createHttpClient(cf *config.NSXOperatorConfig, useExternalHttp bool) *http.Client {
+	var httpClient *http.Client = nil
+	if useExternalHttp {
+		tr := &http.Transport{}
+		transport := &Transport{Base: tr}
+		httpClient = &http.Client{
+			Transport: transport,
+		}
+
+		if cf.EnvoyPort != 0 {
+			//envoy client
+			if len(cf.CaFile) > 0 {
+				extraHeaders := make(map[string]string)
+				extraHeaders["x-vmware-server-tls-cert"] = certPemBytesToHeader(cf.CaFile[0])
```

Comment:
> cf.CaFile[0]

Do we always use the first NSX manager even with NCP cleanup? I assume this is also something yet to be handled on NCP side of cleanup. Apparently it's also possible for wcp to pass per-manager leaf certs. So we need to bring in the positional thumbprint / cert selection implemented in the normal NCP / operator runtime.

They also mentioned a possibility of mixed use of thumbprint and cert (on a per-mgr basis).

Reply:
Now use http client from nsx-operator. so the nsx could be changed if the nsx  is not available.